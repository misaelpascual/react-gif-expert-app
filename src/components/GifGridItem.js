import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__zoomIn animate__delay-1s">
      <img src={ url } alt={ title } />
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string,
  url: PropTypes.string.isRequired
}