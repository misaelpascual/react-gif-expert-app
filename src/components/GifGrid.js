import React from 'react'
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import { LoadingBase } from './UI/LoadingBase'

export const GifGrid = ({ category }) => {

  // const [images, setImages] = useState([]);
  const { data:images, loading } = useFetchGifs(category)


  return (
    <>
      <h3 className='animate__animated animate__fadeInDown animate__delay-1s'> { category } </h3>
      { loading && <LoadingBase /> }
      <div className="card_grid">
        {
          images?.map((img) => (
            <GifGridItem
              key={ img.id }
              { ...img }
            />
          ))
        }
      </div>
    </>
  )
}

GifGrid.protoTypes = {
  category: PropTypes.string.isRequired
}
