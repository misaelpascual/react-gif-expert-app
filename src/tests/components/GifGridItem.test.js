import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('Prueba en el componente <GifGridItem.test />', () => {
  const title = 'Gif'
  const url = 'https://media0.giphy.com/media/hryis7A55UXZNCUTNA/giphy.gif?cid=0db280a0zvvdt67i7zxo8cd1791yuy9jpb8odjgdoaw5bi66&rid=giphy.gif&ct=g'

  const wrapper = shallow(<GifGridItem title={title} url={url} />)

  test('Debe mostrar <GifGridItem.test /> correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Debe de tener la imagen al url y alt de los props', () => {
    const img = wrapper.find('img')

    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)
  })

  test('Debe de tener card', () => {
    const div = wrapper.find('div')

    expect(div.hasClass('card')).toBe(true)
  })

})
