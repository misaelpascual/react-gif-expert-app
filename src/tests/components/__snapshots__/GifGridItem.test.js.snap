// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente <GifGridItem.test /> Debe mostrar <GifGridItem.test /> correctamente 1`] = `
<div
  className="card animate__animated animate__zoomIn animate__delay-1s"
>
  <img
    alt="Gif"
    src="https://media0.giphy.com/media/hryis7A55UXZNCUTNA/giphy.gif?cid=0db280a0zvvdt67i7zxo8cd1791yuy9jpb8odjgdoaw5bi66&rid=giphy.gif&ct=g"
  />
</div>
`;
